jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/JavaContainerBuilderHelper.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.plugins.common;
+
+import com.google.cloud.tools.jib.api.JavaContainerBuilder;
+import com.google.cloud.tools.jib.api.JibContainerBuilder;
+import com.google.cloud.tools.jib.api.RegistryImage;
+import com.google.cloud.tools.jib.configuration.FilePermissions;
+import com.google.cloud.tools.jib.configuration.LayerConfiguration;
+import com.google.cloud.tools.jib.filesystem.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.filesystem.DirectoryWalker;
+import com.google.cloud.tools.jib.filesystem.RelativeUnixPath;
+import com.google.cloud.tools.jib.frontend.JavaLayerConfigurations.LayerType;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/** Helper for constructing {@link JavaContainerBuilder}-based {@link JibContainerBuilder}s. */
+public class JavaContainerBuilderHelper {
+
+  /**
+   * Returns a {@link LayerConfiguration} for adding the extra directory to the container.
+   *
+   * @param extraDirectory the source extra directory path
+   * @param extraDirectoryPermissions map from path on container to file permissions
+   * @return a {@link LayerConfiguration} for adding the extra directory to the container
+   * @throws IOException if walking the extra directory fails
+   */
+  public static LayerConfiguration extraDirectoryLayerConfiguration(
+      Path extraDirectory, Map<AbsoluteUnixPath, FilePermissions> extraDirectoryPermissions)
+      throws IOException {
+    LayerConfiguration.Builder builder =
+        LayerConfiguration.builder().setName(LayerType.EXTRA_FILES.getName());
+    new DirectoryWalker(extraDirectory)
+        .filterRoot()
+        .walk(
+            path -> {
+              AbsoluteUnixPath pathOnContainer =
+                  AbsoluteUnixPath.get("/").resolve(extraDirectory.relativize(path));
+              builder.addEntry(
+                  path, pathOnContainer, extraDirectoryPermissions.get(pathOnContainer));
+            });
```

Comment:
Can't this be the code in the previous `Maven/GradleLayerConfigurations`?

```java
    if (Files.exists(extraDirectory)) {	
      layerBuilder.addDirectoryContents(	
          LayerType.EXTRA_FILES,	
          extraDirectory,	
          path -> true,	
          AbsoluteUnixPath.get("/"),	
          extraDirectoryPermissions);	
    }
```

Reply:
`addDirectoryContents` is a method on `JavaLayerConfigurations`, not `LayerConfiguration.Builder`

Reply:
Ah, right.